code/model/editableformfields/EditableOption.php

```diff
@@ -114,4 +135,19 @@ public function canPublish($member = null) {
     public function canUnpublish($member = null) {
         return $this->canDelete($member);
     }
+
+    /**
+     * Fetches a value for $this->Value. If empty values are not allowed,
+     * then this will return the title in the case of an empty value.
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        $value = $this->getField('Value');
+        if(empty($value) && !self::allow_empty_values()) {
+            return $this->Title;
```

Comment:
No it's good... it's cast to Varchar because it's in $db, which means it gets properly escaped. :D